### ISSUE TITLE
Add ternaries (b ? c : d)

### ISSUE THREAD

**Issue by staycoolcall911** [winglang/wing#221]
We'd like to add support for ternaries.

- [ ] Add to the winglang spec
- [ ] Add to the wingc compiler

**Comment by Chriscbr** [comment]
It would be pretty cool if we could reuse `if / else / elif` in the expression position (like Rust and Swift allows) since IMO it's slightly more readable than the archaic ?: ternary. See this Swift proposal for more details: https://github.com/apple/swift-evolution/blob/main/proposals/0380-if-switch-expressions.md. What do others think?

**Comment by yoav-steinberg** [comment]
I like the idea of turning `if` statements into expressions (like rust). I also think that given the extensive use of the `?` char when dealing with optionals we might run into grammar challenges or just confusion because `?` is usually related to optionals:
`let x = (a ?? b?) ? c?.y : 1; // Explain this`

**Comment by MarkMcCulloh** [comment]
Personally I find ternaries readable and dislike `if` expressions. I also dislike variable shadowing though so I'm probably just a luddite ¯\\_(ツ)_/¯

**Comment by hasanaburayyan** [comment]
I agree with @MarkMcCulloh I find the `if` expressions to make things look very busy but thats just preference.

**Comment by Chriscbr** [comment]
The syntactic ambiguity raised by @yoav-steinberg is valid - fortunately, I think the visual similarity between `expr ? ...` and `expr? ...` should be addressed by removing the unwrap/condition expression syntax (https://github.com/winglang/wing/issues/5942).

The "traditional" ternary syntax `a ? b : c` is already in a lot of languages today (C, C++, Java, JavaScript, PHP, Ruby, Swift, Perl) so I think going forward with this design makes the most sense. One of our design goals is to make Wing to feel familiar to people who have used other popular languages.

Type checking this expression can be modeled after the logic already used for type checking simple if/else statements in the compiler.

I'll mark this as a good issue if anyone wants to pick this up!

**Comment by Pahul-Sidhu** [comment]
Hey Chris, I want to pick this up.

**Comment by Chriscbr** [comment]
Thanks @Pahul-Sidhu! I've assigned the issue to you. You'll want to look into adding ternaries to [grammar.js](https://github.com/winglang/wing/blob/main/libs/tree-sitter-wing/grammar.js) and then incorporate it into the rest of the parser [here](https://github.com/winglang/wing/blob/main/libs/wingc/src/parser.rs). Let me know if I can provide other pointers or help with debugging any confusing errors.

**Comment by Pahul-Sidhu** [comment]
Thank you
